src/harness/harnessUtils.ts

```diff
@@ -169,54 +169,15 @@ namespace Utils {
     export function sourceFileToJSON(file: ts.Node): string {
         return JSON.stringify(file, (_, v) => isNodeOrArray(v) ? serializeNode(v) : v, "    ");
 
-        function getKindName(k: number | string): string {
-            if (ts.isString(k)) {
+        function getKindName(k: number | string | undefined): string | undefined {
+            if (k === undefined || ts.isString(k)) {
                 return k;
             }
-
-            // For some markers in SyntaxKind, we should print its original syntax name instead of
-            // the marker name in tests.
-            if (k === (ts as any).SyntaxKind.FirstJSDocNode ||
-                k === (ts as any).SyntaxKind.LastJSDocNode ||
-                k === (ts as any).SyntaxKind.FirstJSDocTagNode ||
-                k === (ts as any).SyntaxKind.LastJSDocTagNode) {
-                for (const kindName in (ts as any).SyntaxKind) {
-                    if ((ts as any).SyntaxKind[kindName] === k) {
-                        return kindName;
-                    }
-                }
-            }

```

Comment:
Using `ts.Debug.formatSyntaxKind` is slower because it does a lot more work than these 4 hardcoded things. To regain the perf, I have another PR that will make `formatSyntaxKind` faster.

I still think it's better to not write the same enum-member-to-string code more than once, but it's annoying that a perf fix is required to do it.

Reply:
#49487 recovers the performance; it's 25% of the test run :(